### PR TITLE
Address a refactoring concern from #54

### PR DESCRIPTION
**Pull request by RussTheAerialist**
More readable by extracting a couple of methods and reorganizing the code a little.

### CODE REVIEW

File: tableaudocumentapi/datasource.py [hunk 1]
```diff
@@ -178,16 +179,17 @@ def fields(self):
...
-        missing_fields = (x for x in metadata_fields if x not in existing_fields)
-        column_objects.extend((
-            _column_object_from_metadata_xml(_get_metadata_xml_for_field(self._datasourceTree, field_name))
-            for field_name in missing_fields
-        ))
-
-        return FieldDictionary({k: v for k, v in column_objects})
+        column_field_objects = self._get_column_objects()
+        existing_column_fields = [x.id for x in column_field_objects]
+        metadata_field_objects = (x for x in self._get_metadata_objects() if x.id not in existing_column_fields)
```
Comment (t8y8) [inline tableaudocumentapi/datasource.py]: metadata_only_field_objects -- or some hint that we have to find these a different way

**Comment by t8y8** [comment]
How do I make the rocket emoji? (rocket)

**Comment by RussTheAerialist** [comment]
`:rocket:`  I'll rename the variable to metadata_only_field_objects

**Comment by t8y8** [comment]
:rocket:

**Pull request merged**